sdk/src/esplora.ts

```diff
@@ -333,7 +333,11 @@ export class EsploraClient {
      */
     async getFeeEstimate(confirmationTarget: number): Promise<number> {
         const response = await this.getJson<Record<number, number>>(`${this.basePath}/fee-estimates`);
-        return response[confirmationTarget];
+        if (response[confirmationTarget]) return response[confirmationTarget];
+        const TARGET_MAX = 1008;
+        for (let target = confirmationTarget + 1; target <= TARGET_MAX; target++) {
+            if (response[target]) return response[target];
+        }
```

Comment:
I think a better approach is just to convert the `Record` to an array, sort by the confirmation number, and return the first entry in that array.

Reply:
Speaking of time/memory complexity this approach will take O(n) memory + O(n logn) time
Simple for loop takes just O(n) time

Reply:
Fixed. Desc sorting + pop()

Reply:
You were right with the complexity in your comment before btw ;)

Reply:
but actually I'm not so sure we want this need to consider, fee rate on testnet should always be 1